### PR TITLE
Cleaning a few warnings in the test

### PR DESCRIPTION
**Pull request by BoxiLi**
- pytest warning to error
- Use raw string literals for docstrings with LaTeX
- Skip plotting tests correctly when matplotlib not installed

### CODE REVIEW

File: tests/test_device.py [hunk 1]
```diff
@@ -5,6 +5,7 @@
 import warnings
 import numpy as np
 import pytest
+plt = pytest.importorskip("matplotlib.pyplot")
```
Comment (hodgestar) [inline tests/test_device.py:8]: Will this skip all the tests in this file?

Comment (BoxiLi) [reply tests/test_device.py:8]: You are right this skips the whole test module. Importation has been moved inside the test function.

**Review by hodgestar — APPROVED** [review]

**Review by hodgestar — APPROVED** [review]

**Comment by BoxiLi** [comment]
Thank you @hodgestar , just narrowed down the range of this warning filter a bit more.

**Review by hodgestar — APPROVED** [review]
Still looks good to me. Thank you for cleaning this up!

**Pull request merged**